modules/uolBidAdapter.js

```diff
@@ -0,0 +1,202 @@
+        utils.logError(UOL_LOG_HEADER + 'Invalid param definition for cpmFactor on bidID ' + bid.bidId);
+        isValid = false;
+      }
+    } else {
+      isValid = false;
+      utils.logError(UOL_LOG_HEADER + 'No params defined for bidID ' + bid.bidId);
+    }
+    if (getSizes(bid).length == 0) {
+      utils.logError(UOL_LOG_HEADER + 'No sizing definition found for bidID ' + bid.bidId);
+      isValid = false;
+    }
+    return isValid;
+  },
+  /**
+    * Make a server request from the list of BidRequests.
+    *
+    * @param {validBidRequests[]} - an array of bids
+    * @param {bidderRequests} - an object containing all bid params, including validBids.
+    * @return ServerRequest Info describing the request to the server.
+    */
+  buildRequests: function(validBidRequests, bidderRequest) {
+    var data = JSON.stringify(getPayloadFor(validBidRequests, bidderRequest));
+    return {
+      method: 'POST',
+      url: ENDPOINT_URL,
+      data,
+      bidRequest: validBidRequests,
+    };
+  },
+  /**
+     * Unpack the response from the server into a list of bids.
+     *
+     * @param {ServerResponse} serverResponse A successful response from the server.
+     * @return {Bid[]} An array of bids which were nested inside the server.
+     */
+  interpretResponse: function(serverResponse, {bidRequest}) {
+    const bidResponses = [];
+    if (serverResponse.body) {
+      const ads = serverResponse.body.ads;
+      for (var index = 0; index < ads.length; index++) {
+        const bidResponse = {
+          requestId: ads[index].bidId,
+          cpm: ads[index].cpm,
+          width: ads[index].width,
+          height: ads[index].height,
+          creativeId: ads[index].creativeId,
+          dealId: ads[index].dealId,
+          currency: ads[index].currency,
+          netRevenue: ads[index].netRevenue,
+          ttl: ads[index].ttl,
+          ad: ads[index].ad,
+          mediaType: ads[index].mediaType
+        };
+        bidResponses.push(bidResponse);
+      }
+    }
+    return bidResponses;
+  },
+
+  /**
+     * Register the user sync pixels which should be dropped after the auction.
+     *
+     * @param {SyncOptions} syncOptions Which user syncs are allowed?
+     * @param {ServerResponse[]} serverResponses List of server's responses.
+     * @return {UserSync[]} The user syncs which should be dropped.
+     */
+  getUserSyncs: function(syncOptions, serverResponses) {
+    const syncs = [];
+    if (syncOptions.iframeEnabled && serverResponses.body.trackingPixel) {
+      syncs.push({
+        type: 'iframe',
+        url: serverResponses.body.trackingPixel
+      });
+    }
+    return syncs;
+  },
+
+  /**
+    * Register bidder specific code, which will execute if bidder timed out after an auction
+    * @param {data} Containing timeout specific data
+    */
+  onTimeout: function(data) {
+    // Bidder specifc code
+    return true;
```

Comment:
Is there any more you would wish to do in the case of a timeout?

Reply:
For the time being we have no plans hence, Im removing the method call.